File: backend/src/main/kotlin/com/budgetbook/feedback/domain/FeedbackComment.kt
package com.budgetbook.feedback.domain

import com.budgetbook.auth.domain.User
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import org.springframework.data.annotation.CreatedDate
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "feedback_comments")
class FeedbackComment(
    @Id
    val id: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    val post: FeedbackPost,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    val author: User,

    @Column(nullable = false, columnDefinition = "TEXT")
    val content: String,

    @Column(name = "is_admin_reply", nullable = false)
    val isAdminReply: Boolean = false,

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    val createdAt: Instant = Instant.now()
)

File: frontend/lib/features/feedback/domain/entities/feedback_comment.dart
import 'package:equatable/equatable.dart';

class FeedbackComment extends Equatable {
  final String id;
  final String authorId;
  final String authorName;
  final String content;
  final bool isAdminReply;
  final DateTime createdAt;

  const FeedbackComment({
    required this.id,
    required this.authorId,
    required this.authorName,
    required this.content,
    required this.isAdminReply,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        authorId,
        authorName,
        content,
        isAdminReply,
        createdAt,
      ];
}
